pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -0,0 +1,1089 @@
+
+		dst.Spec.Pravega.LongTermStorage = &LongTermStorageSpec{
+			FileSystem: &FileSystemSpec{
+				PersistentVolumeClaim: srcObj.Spec.Pravega.Tier2.FileSystem.PersistentVolumeClaim,
+			},
+		}
+	} else if srcObj.Spec.Pravega.Tier2.Ecs != nil {
+		dst.Spec.Pravega.LongTermStorage = &LongTermStorageSpec{
+			Ecs: &ECSSpec{
+				ConfigUri:   srcObj.Spec.Pravega.Tier2.Ecs.ConfigUri,
+				Bucket:      srcObj.Spec.Pravega.Tier2.Ecs.Bucket,
+				Prefix:      srcObj.Spec.Pravega.Tier2.Ecs.Prefix,
+				Credentials: srcObj.Spec.Pravega.Tier2.Ecs.Credentials,
+			},
+		}
+	} else if srcObj.Spec.Pravega.Tier2.Hdfs != nil {
+		dst.Spec.Pravega.LongTermStorage = &LongTermStorageSpec{
+			Hdfs: &HDFSSpec{
+				Uri:               srcObj.Spec.Pravega.Tier2.Hdfs.Uri,
+				Root:              srcObj.Spec.Pravega.Tier2.Hdfs.Root,
+				ReplicationFactor: srcObj.Spec.Pravega.Tier2.Hdfs.ReplicationFactor,
+			},
+		}
+	}
+
+	// Controller Resources
+	if srcObj.Spec.Pravega.ControllerResources != nil {
+		dst.Spec.Pravega.ControllerResources = &corev1.ResourceRequirements{
+			Requests: srcObj.Spec.Pravega.ControllerResources.Requests,
+			Limits:   srcObj.Spec.Pravega.ControllerResources.Limits,
+		}
+	}
+	// SegmentStore Resources
+	if srcObj.Spec.Pravega.SegmentStoreResources != nil {
+		dst.Spec.Pravega.SegmentStoreResources = &corev1.ResourceRequirements{
+			Requests: srcObj.Spec.Pravega.SegmentStoreResources.Requests,
+			Limits:   srcObj.Spec.Pravega.SegmentStoreResources.Limits,
+		}
+	}
+	return nil
+}
+
+func (dst *PravegaCluster) ConvertFrom(srcRaw conversion.Hub) error {
+	log.Printf("Converting Pravega CR version from v1alpha1 to v1beta1.")
+	//logic for conveting from v1alpha1 to v1beta1
+	srcObj := srcRaw.(*v1alpha1.PravegaCluster)
+	dst.ObjectMeta = srcObj.ObjectMeta
+	err := dst.convertSpec(srcObj)
+	if err != nil {
+		log.Fatalf("Error converting CR object from version v1alpha1 to v1beta1 %v", err)
+		return err
+	}
+	err = dst.migrateBookkeeper(srcObj)
+	if err != nil {
+		return err
+	}
+	err = dst.updatePravegaOwnerReferences()
+	if err != nil {
+		return err
+	}
+	log.Print("Version migration completed successfully.")
+	return nil
+}
+
+func (dst *PravegaCluster) updatePravegaOwnerReferences() error {
+	ownerRefs := &[]metav1.OwnerReference{
+		{
+			APIVersion:         "pravega.pravega.io/v1beta1",
+			Kind:               "PravegaCluster",
+			Name:               dst.GetName(),
+			UID:                dst.GetUID(),
+			BlockOwnerDeletion: pointer.BoolPtr(true),
+			Controller:         pointer.BoolPtr(true),
+		},
+	}
+	err := dst.updateSSSReferences(*ownerRefs)
+	if err != nil {
+		return err
+	}
+	log.Print("Updated SegmentStore OwnerReferences.")
+	err = dst.updateControllerReferences(*ownerRefs)
+	if err != nil {
+		return err
+	}
+	log.Print("Updated Controller OwnerReferences.")
+	return nil
+}
+
+func (dst *PravegaCluster) updateSSSReferences(ownerRefs []metav1.OwnerReference) error {
+	if util.IsVersionBelow07(dst.Spec.Version) {
```

Comment:
why are we updating owner references for ss pvcs only for versions below 07 and how the pvcs are teansferred in case of above 07?